### PR TITLE
Add support for --cache-from

### PR DESCRIPTION
**Pull request by Chippiewill**
This adds support for `docker build` style --cache-from. It achieves
this with two changes:

1. img will _always_ ask buildkit to export the build cache inline
2. img will pass through --cache-from arguments

 #205

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -255,6 +257,26 @@ func (cmd *buildCommand) Run(args []string) (err error) {
...
+
+	var cacheImports []*controlapi.CacheOptionsEntry;
+	for _, cacheTarget := range cmd.cacheFrom.GetAll() {
+		cacheImports = append(cacheImports, &controlapi.CacheOptionsEntry{
+			Type:  "registry",
+			Attrs: map[string]string{
+				"ref": cacheTarget,
+			},
+		});
+    }
```
Comment (Chippiewill) [inline build.go]: I do not understand why this is insufficient by itself and still requires the `cache-from` frontendAttrs to be passed through

File: build.go [hunk 2]
```diff
@@ -255,6 +257,26 @@ func (cmd *buildCommand) Run(args []string) (err error) {
 	eg.Go(func() error {
 		return sess.Run(ctx, sessDialer)
 	})
+
+	var cacheExports []*controlapi.CacheOptionsEntry;
+	cacheExports = append(cacheExports, &controlapi.CacheOptionsEntry{
+		Type:  "inline",
```
Comment (Chippiewill) [inline build.go]: It seems non-ideal to set it to always inline, my logic is that this should follow as closely as possible the behavior of `docker build`. I suppose another option would be to default it to on, but let it be configurable - or even to expose the full buildctl style export cache arguments to allow exporting to separate images.

I'm not really sure what the overhead of the cache in the image is. If it's small enough I'd be inclined to leave it.

**Comment by AkihiroSuda** [comment]
```
go build -tags "seccomp" -ldflags "-w -X github.com/genuinetools/img/version.GITCOMMIT=b5155e9 -X github.com/genuinetools/img/version.VERSION=v0.5.7" -o img .
client/controller.go:7:2: cannot find package "github.com/moby/buildkit/cache/remotecache/inline" in any of:
	/home/travis/gopath/src/github.com/genuinetools/img/vendor/github.com/moby/buildkit/cache/remotecache/inline (vendor tree)
	/home/travis/.gimme/versions/go1.11.13.linux.amd64/src/github.com/moby/buildkit/cache/remotecache/inline (from $GOROOT)
	/home/travis/gopath/src/github.com/moby/buildkit/cache/remotecache/inline (from $GOPATH)
client/controller.go:8:2: cannot find package "github.com/moby/buildkit/cache/remotecache/registry" in any of:
	/home/travis/gopath/src/github.com/genuinetools/img/vendor/github.com/moby/buildkit/cache/remotecache/registry (vendor tree)
	/home/travis/.gimme/versions/go1.11.13.linux.amd64/src/github.com/moby/buildkit/cache/remotecache/registry (from $GOROOT)
```

needs revendor

**Comment by AkihiroSuda** [comment]
Ideally we should support non-registry cache as well, but can be separate PR

File: build.go (hunk 2 above)
Comment (yangm97) [reply build.go]: I was looking forward using `mode=max` for `--export-cache` since it exports multi-stage build layers.

**Pull request merged**